tools/render-test/render-test-main.cpp

```diff
@@ -400,29 +399,38 @@ SlangResult _assignVarsFromLayout(
     return context.assign(rootCursor, layout.rootVal);
 }
 
-void RenderTestApp::applyBinding(PipelineType pipelineType, ICommandEncoder* encoder)
+Result RenderTestApp::applyBinding(PipelineType pipelineType, ICommandEncoder* encoder)
 {
+    auto slangReflection = (slang::ProgramLayout*)spGetReflection(
+        m_compilationOutput.output.getRequestForReflection());
```

Comment:
Hmm... I don't really like us having to add this bit. What do we need the layout for here, and is it possible that we can just query the layout from the root shader object instead?

Reply:
I looked at the relevant code to see why this comes up, and it vaguely makes sense because the `//TEST_INPUT` lines may allocate shader objects and thus need to be able to look up types by name. This code here doesn't feel like the appropriate place to be querying the program layout, but I guess its is reasonable for now.

Reply:
This is what is already there and not newly added in this PR. It is definitely good to figure out how to simplify them.